### PR TITLE
Initial Commit

### PR DESCRIPTION
**Pull request by Ayo-lextego**

### CODE REVIEW

File: src/controllers/execute.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  };
+  for (const channel of req.networkMap.messages[0].channels) {
+    for (const typology of channel.typologies) {
+      if (typology.rules.some((rule) => rule.id === ruleResult.id)) {
+        const typologyResponse = await axios.post(`${typology.host}/execute`, toSend);
+        if (typologyResponse.status !== 200) {
+          LoggerService.error(typologyResponse.data);
+        }
+      }
+    }
```
Comment (vorsterk) [inline src/controllers/execute.ts:64]: Please can you add a return statement , the for (const typology of channel.typologies) to leave the block , It was Foley's Suggested change .

File: src/toMobileNumber.d.ts [hunk 2]
```diff
@@ -0,0 +1,3 @@
+declare interface String {
+  toMobileNumber(): string;
```
Comment (vorsterk) [inline src/toMobileNumber.d.ts:2]: Is this needed?

**Review by vorsterk — CHANGES_REQUESTED** [review]

**Review by johanfol — CHANGES_REQUESTED** [review]

File: .editorconfig [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+
+[*]
+indent_style = space
+indent_size = 2
+charset = utf-8
+trim_trailing_whitespace = true
+insert_final_newline = true
+
+[*.md]
+trim_trailing_whitespace = false
```
Comment (uche-lextego) [inline .editorconfig:11]: ADD end of file line here.

File: .env.template [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+process.env.APM_URL=''
+process.env.APM_SECRET_TOKEN=''
+process.env.NODE_ENV=''
+process.env.DATABASE_NAME=''
+process.env.COLLECTION_NAME=''
+process.env.DATABASE_URL=''
+process.env.DATABASE_USER=''
+process.env.DATABASE_PASSWORD=''
+process.env.CONFIG_DATABASE=''
+process.env.CONFIG_COLLECTION=''
```
Comment (uche-lextego) [inline .env.template:15]: Add end of file line here.

File: .eslintignore [hunk 5]
```diff
@@ -0,0 +1,5 @@
+coverage
+templates
+build
+**/node_modules
+*.spec.ts
```
Comment (uche-lextego) [inline .eslintignore:5]: Add end of file line here

File: package.json [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  "description": "Actio FRM Rule 030",
+  "main": "src/index.ts",
+  "types": "build/index.d.ts",
+  "scripts": {
+    "dev": "nodemon",
+    "build": "tsc --project tsconfig.json",
+    "cleanup": "rm -rf build template jest.config.js jest.config.js.map node_modules yarn.lock",
+    "start": "node build/index",
+    "start:dev": "env $(cat .env.template) nodemon",
+    "fix": "yarn fix:prettier && yarn fix:eslint",
```
Comment (uche-lextego) [inline package.json:13]: Change everything Yarn on this file to npm run. We don;t use yarn again

File: package.json [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+    "fix:eslint": "eslint --fix \"**/*.ts\"",
+    "fix:prettier": "prettier --write \"**/*.ts\"",
+    "lint": "yarn lint:eslint && yarn lint:prettier",
```
Comment (uche-lextego) [inline package.json:16]: Here too

File: rule-030-deploy.yml [hunk 8]
```diff
@@ -0,0 +1,15 @@
...
+  off-rule-030-rel-1-0-0:
+    namespace: openfaas-fn
+    lang: dockerfile
+    handler: .
+    image: actio.azurecr.io/off-rule-030-rel-1-0-0:latest
+    annotations:
+      com.openfaas.health.http.path: /health
+      com.openfaas.health.http.initialDelay: "45s"
+    secrets:
+      - frmpullsecret
```
Comment (uche-lextego) [inline rule-030-deploy.yml:15]: End of file line here too

File: src/toMobileNumber.d.ts (hunk 2 above)
Comment (uche-lextego) [reply src/toMobileNumber.d.ts:2]: Nope, @Ayo-lextego please remove this file.

**Review by uche-lextego — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
...ifQ.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJmcm0iLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlY3JldC5uYW1lIjoiYXBtLXNlcnZlci1hcG0tc2VydmVyLXRva2VuLXpkaHNzIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQubmFtZSI6ImFwbS1zZXJ2ZXItYXBtLXNlcnZlciIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VydmljZS1hY2NvdW50LnVpZCI6IjI3NzhjY2Q3LTMwMjEtNDdhMy04YTU1LTdiYmJhMGQ1ZjQ3ZCIsInN1YiI6InN5c3RlbTpzZXJ2aWNlYWNjb3VudDpmcm06YXBtLXNlcnZlci1hcG0tc2VydmVyIn0.UWqRaNxzBCqHPnzQGiHd2BPk3Gsi81C7_Al3HdYWnfxFmqh3ng1SaAXKm0pNH8OUESiUHI9p0dftJr3A5AgtKALMMIl2OK4oFYVeIgfKYiZ5WYCeiaawLFg6sRph0uYZsfTvhDNmy2NaEElLpIsT7_8XXzc12NCZvIJ6sKL5Ue1GBClSu1avEGKHzq1TNzP2drxsBU-BLJGaXOrTHvJp4w1J-NXKlf0Xluq3eMF-8TLj3Hx06i7pcHi5yGPFGSCf_qu1jPjh-ValPJUfDb9UIQZVepbhuzvI10VXt0JXlUw-nc7yb66SqKvwAp4_W92dYjrP3PeExpfkrGP4-0yzaAzTDocCUKHejO-inI5H16N0SNaGrGTktGj9HK6F_PXGMRqvHCzJKhi6rUpkf_WtcjvKEBlVvLCdsTcgfQOA_JyK-H0-E0yFIhQgN7t0ZaQG_mtduxniG4hIZszwuJzUdEftccOXlBwEdsLncmqxOCZe9epSQ8zqCMR-B4O1W3eBMs8yCzvNz9Zf2GK-gCdJ-ZvUU8JSM6GLnlBj5zBExtBnsc525W6SbxPOTMQsiAjymhLNOgvphfOIiwVHDK1F2QYuSOWDAa9nX8XOBRoc3uqmPnPx8FE3WW5nzzQff_MrAALxDCnMIUgjWpurT-4puH1u1No-jfQzdgb9JPkQiz0"
```
Comment (johanfol) [inline Dockerfile:53]: Please don't commit sensitive information.

File: src/controllers/execute.ts [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    ruleResult: ruleResult,
+    networkMap: req.networkMap,
+  };
+  for (const channel of req.networkMap.messages[0].channels) {
+    for (const typology of channel.typologies) {
+      if (typology.rules.some((rule) => rule.id === ruleResult.id)) {
+        const typologyResponse = await axios.post(`${typology.host}/execute`, toSend);
+        if (typologyResponse.status !== 200) {
+          LoggerService.error(typologyResponse.data);
+          return;
```
Comment (johanfol) [inline src/controllers/execute.ts:62]: This return is in the wrong location - you're only stopping to send requests to the TP IF there was an issue? That doesn't make sense. Please move this return just one line down (to be inside the oarent IF, but outside the current IF.

File: src/services/logic.service.ts [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    result: false,
+    subRuleRef: '',
+    reason: '',
+  };
+
+
+  // e2e ID
+  const endToEndId = req.transaction.FIToFIPmtSts.TxInfAndSts.OrgnlEndToEndId;
+  // rule config from DB
+  const sRuleConfig = await dbService.getRuleConfig(ruleResult.id);
```
Comment (johanfol) [inline src/services/logic.service.ts:21]: We should be using id and CFG to get rule config.

**Review by johanfol — CHANGES_REQUESTED** [review]

**Review by vorsterk — APPROVED** [review]

File: src/services/logic.service.ts [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+export const handleTransaction = async (req: RuleRequest): Promise<RuleResult> => {
+  const ruleResult: RuleResult = {
+    id: `030@${config.ruleVersion}`,
+    cfg:  '',
+    result: false,
+    subRuleRef: '',
+    reason: '',
+  };
+
+  for (const messages of req.networkMap.messages) {
```
Comment (johanfol) [inline src/services/logic.service.ts:17]: Please see comment on Jude's PR, same applies here - just check in with him and Kyle please.

**Review by johanfol — CHANGES_REQUESTED** [review]

**Pull request merged**